src/GTP.cpp

```diff
@@ -544,10 +544,13 @@ bool GTP::execute(GameState & game, std::string xinput) {
         cmdstream >> tmp;   // eat place_free_handicap
         cmdstream >> stones;
 
-        game.place_free_handicap(stones);
-
-        std::string stonestring = game.board.get_stone_list();
-        gtp_printf(id, "%s", stonestring.c_str());
+        if (!cmdstream.fail()) {
+            game.place_free_handicap(stones);
+            std::string stonestring = game.board.get_stone_list();
+            gtp_printf(id, "%s", stonestring.c_str());
+        } else {
+            gtp_fail_printf(id, "Not a valid number of handicap stones");
```

Comment:
I think this is the wrong error message in this case. "Not a valid list of vertices" or something similar?

Reply:
If you look five lines above you'll see it reads an integer "stones" from cmdstream, so I think this is the right error message.

Also see https://www.gnu.org/software/gnugo/gnugo_19.html
"Fails:     invalid number of stones"